src/main/java/uk/gov/service/notify/RetentionPeriodDuration.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.service.notify;
+
+import java.time.temporal.ChronoUnit;
+import java.util.Locale;
+
+/***
+ * Represents a retention period duration for files for file upload
+ */
+public class RetentionPeriodDuration {
+    private final int amount;
+    private final ChronoUnit unit;
+
+    /***
+     * Creates an employee with the specified name and
+     * @param amount - The amount of units of time. Example: If unit is ChronoUnit.WEEKS, and amount is 52,
+     *               then the duration is 52 weeks.
+     * @param unit - The unit of time that is being represented as a ChronoUnit. Currently only supports
+     *             ChronoUnit.WEEKS until API changes allow for other unit types
+     */
+    public RetentionPeriodDuration(int amount, ChronoUnit unit) {
+        this.amount = amount;
+        this.unit = unit;
+        validate();
+    }
+
+    public int getAmount() {
+        return amount;
+    }
+
+    public ChronoUnit getUnit() {
+        return unit;
+    }
+
+    /***
+     * Validates the object; this should be run when the object is constructed
+     * @throws IllegalArgumentException - This is thrown whenever either amount or unit are considered invalid
+     */
+    public void validate(){
+        if(this.unit == ChronoUnit.WEEKS){
+            if(this.amount < 1 || this.amount > 78){
+                throw new IllegalArgumentException(
+                        "When unit is ChronoUnit.WEEKS, value must be larger than 0 and smaller than 79"
+                );
+            }
+        } else {
+            // Currently only Weeks is supported by the API; this can be updated when
+            // API changes allows for different values, or specifically when DAYS is added (as then we can
+            // convert times to days with ChronoUnit
+            throw new IllegalArgumentException(
+                    String.format(
+                            "%s is not a valid unit. Only ChronoUnit.WEEKS is currently supported.",
+                            this.unit.name()
+                    )
+            );
+        }
+    }
```

Comment:
I think we'd want to drop this check out and let the API handle it (otherwise we're doing these checks in two places which will make it trickier/more error-prone to update in the future)

Reply:
I've pushed out a change that removed the validation checks as it's not particularly important for this PR's goals. 

I would perhaps suggest having a think about it in the future. For me, in-class validation has been pretty useful on my current project in preventing mistakes (and it doesn't have to be super strict - just there to rule out the verifiably insane stuff), and you _could_ argue it would reduce load on the API by catching incorrect stuff before it hits it. I do appreciate it's an extra work load though 😅

Reply:
Yep, definitely pros and cons either way and it's not a strongly held opinion for now.